### DISCUSSION TITLE
Fatal Error: Unexpected argument 'cudaplot'

### DISCUSSION THREAD

**Discussion by whattheflux** [guydavis/machinaris#874] [General]
Machinaris 0.8.7 on unraid 
tesla p4
configured bladebit with gpuplot in settings
when I tail the logs i get the above error:
Fatal Error:
Unexpected argument 'cudaplot'

do I need to update further or change branch or something to get on the latest alpha release? Also does this version use the GPU for farming? and can it farm bladebit compressed plots? Thanks.

**Comment by guydavis** [comment]
Greetings, sorry for the issue.  Bladebit is currently undergoing a lot of changes as they try to catch up with Madmax's Gigahorse.  As such, they are dropping new versions quite frequently.  While many are already plotting and farming via GPU with [Gigahorse](https://github.com/guydavis/machinaris/wiki/Gigahorse), compressed plots created with Bladebit are not currently farmable.  As such, they are, at best, a tech preview.

To continue testing Bladebit, you are best to switch to the `machinaris:test` image to get [the most recent fixes](https://github.com/guydavis/machinaris/wiki/Releases#release-streams) happening in Bladebit.  Then check out the Bladebit channel in our [Discord](https://discord.gg/mX4AtMTt87) for the latest.  Hope this helps.

**Reply by whattheflux** [reply]
thanks mate. I appreciate your quick response. I love how active you are with the development, it's amazing! Got it working on the test branch. Was a little nervous, as during the ram allocation machinaris does not update, and tailing the logs didn't do anything. The only way I knew that anything was happening was I could see my RAM utilization increasing in unraid. I was about to cancel the plotting, maybe a thought to increase the output during that time?

**Reply by guydavis** [reply]
Good to hear.  Just to be clear, you did see a plotting job appear in the table on the [Plottings | Jobs page](https://github.com/guydavis/machinaris/wiki/Plotman#plotting-view) though right?  Should happen within about 30 seconds, with the spinning indicator going after click the Start button.

**Reply by whattheflux** [reply]
so nothing happened during the ram allocation. the plotting page remained blank for about 2 minutes while the allocation occured. after my ram utilization hit about 55% on the unraid dashboard, THEN it update and added the plotting job which was about 1-2 minutes. It also adds about that time because I guess it plots as -n1 then has to reallocate for the next job?

**Reply by guydavis** [reply]
Interesting, I would not have expected a two minute pause.  Possibly the ~30 second window is being exceeded for Plotman to complete the launch.  Good to know.

Thanks for the feedback, I'll add support to trigger one `bladebit_cuda` process and have it generate multiple plots by setting the optional `n` value, same as [currently exists for Madmax](https://github.com/guydavis/machinaris/wiki/MadMax#n-plots).
